include/aws/auth/signing_config.h

```diff
@@ -25,79 +25,144 @@ struct aws_credentials;
 
 typedef bool(aws_should_sign_param_fn)(const struct aws_byte_cursor *name, void *userdata);
 
-/*
+/**
  * A primitive RTTI indicator for signing configuration structs
  *
  * There must be one entry per config structure type and it's a fatal error
  * to put the wrong value in the "config_type" member of your config structure.
  */
 enum aws_signing_config_type { AWS_SIGNING_CONFIG_AWS = 1 };
 
-/*
+/**
  * All signing configuration structs must match this by having
  * the config_type member as the first member.
  */
 struct aws_signing_config_base {
     enum aws_signing_config_type config_type;
 };
 
-/*
+/**
  * What version of the AWS signing process should we use.
  */
 enum aws_signing_algorithm {
     AWS_SIGNING_ALGORITHM_V4,
 };
 
-/*
- * How should the signing process transform the request.
+/**
+ * What sort of signature should be computed from the signable?
  */
-enum aws_signing_request_transform {
-    AWS_SRT_HEADER,
-    AWS_SRT_QUERY_PARAM,
+enum aws_signature_type {
+    /**
+     * A signature for a full http request should be computed, with header updates applied to the signing result.
+     */
+    AWS_ST_HTTP_REQUEST_HEADERS,
+
+    /**
+     * A signature for a full http request should be computed, with query param updates applied to the signing result.
+     */
+    AWS_ST_HTTP_REQUEST_QUERY_PARAMS,
+
+    /**
+     * Compute a signature for a payload chunk.  The signable's input stream should be the chunk data and the
+     * signable should contain the most recent signature value (either the original http request or the most recent
+     * chunk) in the "previous-signature" property.
+     */
+    AWS_ST_HTTP_REQUEST_CHUNK,
+
+    /**
+     * Compute a signature for an event stream event.  The signable's input stream should be the event payload, the
+     * signable should contain the most recent signature value (either the original http request or the most recent
+     * event) in the "previous-signature" property as well as any event headers that should be signed with the
+     * exception of ":date"
+     *
+     * This option is not yet supported.
+     */
+    AWS_ST_HTTP_REQUEST_EVENT,
 };
 
-enum aws_body_signing_config_type {
-    AWS_BODY_SIGNING_OFF,
-    AWS_BODY_SIGNING_ON,
-    AWS_BODY_SIGNING_UNSIGNED_PAYLOAD,
+/**
+ * When creating the canonical request, what should go in the body value part?
+ */
+enum aws_signed_body_value_type {
+    /**
+     * Use the sha256 of the empty string.
+     */
+    AWS_SBVT_EMPTY,
```

Comment:
I'm not sure that EMPTY should be an option?

I'm not a aware of a situation where you'd select EMPTY to ignore an actual payload.

Reply:
All the time actually.  For example, some/most? SDKs  disable body signing (and they'd use this option to do so) when the connection is over TLS.

Reply:
OK
Cool
I am a signing newb